src/native/libs/System.Net.Security.Native.Apple/pal_networkframework.m

```diff
@@ -0,0 +1,433 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_networkframework.h"
+#include <Foundation/Foundation.h>
+
+static WriteCallback _writeFunc;
+static ReadCallback _readFunc;
+static StatusUpdateCallback _statusFunc;
+static nw_protocol_definition_t _framerDefinition;
+static nw_protocol_definition_t _tlsDefinition;
+static dispatch_queue_t _tlsQueue;
+static dispatch_queue_t _inputQueue;
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunguarded-availability-new"
+
+PALEXPORT nw_connection_t AppleNetNative_NwCreateContext(int32_t isServer)
+{
+    if (isServer != 0)  // only client supported at this point.
+        return NULL;
+
+    nw_parameters_t nw_parameters = nw_parameters_create_secure_udp(NW_PARAMETERS_DISABLE_PROTOCOL, NW_PARAMETERS_DEFAULT_CONFIGURATION);
+    nw_endpoint_t nw_endpoint = nw_endpoint_create_host("127.0.0.1", "42");
```

Comment:
Is this address and port? Why "42"? And only localhost?

Reply:
Network.framework doesn't support doing TLS without underlying network connection. The whole trick relies on an abuse where the connection itself is created and never used. A "framer" is layer on top of it and then the TLS on top of the framer. The framer then takes the raw TLS data which we expose to `SslStream` and they never reach the underlying (dummy) connection.

It doesn't particularly matter when values are used here as long as they are syntactically and semantically valid. I think it may be better to use a Unix domain socket with some temporary path though.